analysis/plot.py

```diff
@@ -0,0 +1,62 @@
+"""Plot aggregated event counts.
+"""
+import itertools
+import textwrap
+
+import pandas
+from matplotlib import pyplot
+
+from analysis import utils
+
+
+def main():
+    d_in = utils.OUTPUT_DIR / "aggregate"
+    by_day = read(d_in / "sum_by_day.csv.gz")
+    by_week = read(d_in / "mean_by_week.csv.gz")
+
+    d_out = utils.OUTPUT_DIR / "plot"
+    utils.makedirs(d_out)
+
+    figs_cols = plot(by_day, by_week)
+    for fig, col in figs_cols:
+        f_stem = utils.slugify(col)
+        fig.savefig(d_out / f"{f_stem}.png")
+
+
+def read(f_in):
+    date_col = "event_date"
+    return pandas.read_csv(f_in, parse_dates=[date_col], index_col=[date_col])
+
+
+def plot(by_day, by_week):
```

Comment:
currently we have no tests for the plot, which is probably also hard to implement. would it make sense to check simple things about the data, e.g., that we get `min_ts` and `max_ts` right or is this too obvious and not a good way to test the plot function?

Reply:
What would we test? We could break out `min_ts` and `max_ts`, and test them, but their implementations aren't complicated. We could compare observed plots with expected plots, with [pytest-mpl][1], but I'm unsure whether doing so would be worthwhile.

[1]: https://github.com/matplotlib/pytest-mpl